osquery/tables/system/linux/system_info.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdio.h>
+
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genSystemInfo(QueryContext &context) {
+  QueryData results;
+  Row r;
+  r["hostname"] = osquery::getHostname();
+  r["computer_name"] = r["hostname"];
+
+  std::string uuid;
+  r["uuid"] = (osquery::getHostUUID(uuid)) ? uuid : "";
+
+  auto qd = SQL::selectAllFrom("cpuid");
+  for (const auto& row : qd) {
+    if (row["feature"] == "product_name") {
+      r["cpu_brand"] = row["value"];
+    } else if (row["feature"] == "serial") {
+      // A hardware serial is the equivalent of the OS X "cpu_serial".
+      // A CPU serial number is normally not accessible, the returned value
+      // will be consistent, but not correct.
+      r["cpu_serial"] = row["serial"];
+    }
+  }
+
+  // Can parse /proc/cpuinfo or /proc/meminfo for this data.
```

Comment:
Might also be possible to use [`sysconf()`](http://man7.org/linux/man-pages/man3/sysconf.3.html) instead of parsing `/proc/cpuinfo` and `/proc/meminfo`.

`sysconf` does have the following caveat though:
```
These values also exist, but may not be standard.

        - _SC_PHYS_PAGES
              The number of pages of physical memory.  Note that it is
              possible for the product of this value and the value of
              _SC_PAGESIZE to overflow.

        - _SC_AVPHYS_PAGES
              The number of currently available pages of physical memory.

        - _SC_NPROCESSORS_CONF
              The number of processors configured.

        - _SC_NPROCESSORS_ONLN
              The number of processors currently online (available).
```